File: SDKs/amazonaws.com/ec2/2016-11-15/python/src/sdk/models/shared/associatedrole.py
from dataclasses import dataclass, field
from typing import Optional


@dataclass
class AssociatedRole:
    r"""AssociatedRole
    Information about the associated IAM roles.
    """
    
    associated_role_arn: Optional[str] = field(default=None)
    certificate_s3_bucket_name: Optional[str] = field(default=None)
    certificate_s3_object_key: Optional[str] = field(default=None)
    encryption_kms_key_id: Optional[str] = field(default=None)
    

File: SDKs/amazonaws.com/ec2/2016-11-15/typescript/src/sdk/models/shared/associatedrole.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// AssociatedRole
/** 
 * Information about the associated IAM roles.
**/
export class AssociatedRole extends SpeakeasyBase {
  @SpeakeasyMetadata()
  associatedRoleArn?: string;

  @SpeakeasyMetadata()
  certificateS3BucketName?: string;

  @SpeakeasyMetadata()
  certificateS3ObjectKey?: string;

  @SpeakeasyMetadata()
  encryptionKmsKeyId?: string;
}
